### ISSUE TITLE
TM monthly reports

### ISSUE THREAD

**Issue by willemarcel** [hotosm/tasking-manager#3156]
Generate some reports monthly with data about how Tasking Manager is performing in terms of user retention, tasks mapped/validated, area mapped, amount of time users spent...

Some ideas:
- Number of new users
- Number of new users who mapped at least one task
- Number of tasks mapped by new users
- Number of tasks mapped and validated
- Number of users who mapped at least one task
- Number of users who validated at least one task
- Average mapping time
- Average validation time

**Comment by ramyaragupathy** [comment]
This is a great plan @willemarcel -  We should make this functionality available for:
* all projects and
* a specific set of projects which could be used for Project Managers to track progress

**Comment by bopercival-hot** [comment]
Yep, great idea indeed. This is also, exactly the kind of information that I think will contribute to the M&E metrics for the Audacious project, so I think it will be great to get a jump on this. Where will it be output to?

**Comment by bopercival-hot** [comment]
> This is a great plan @willemarcel - We should make this functionality available for:
> 
> * all projects and
> * a specific set of projects which could be used for Project Managers to track progress

Would this also align to the 'Dashboard' work that we talked about for 4.1?

**Comment by Hagellach37** [comment]
Hey @willemarcel @ramyaragupathy ,
this is Benni from Heidelberg.

@bopercival-hot pointed me to this issue since we had some OSM stats, tasking manager stats dicussions in the last weeks. Your ideas seem to align quite well to what we've been doing from the database you provided to us.

Not sure if you are aware of this draft website: http://mapswipe-dev.heigit.org:8086/mapping.html

It basically outputs most of the stats you listed above.

I think this might be a good to have a chat about your ideas and see how the things we've been working on can be integrated.

As a starting point: where are you planning would these analysis live? as a part of the tasking manager code or as a separate project?

**Comment by willemarcel** [comment]
Hi @Hagellach37 

I don't know yet how we'll implement it. The best way would be having it as a TM view and giving the possibility to the user to select different time periods, but I suppose some queries heavy a huge impact on the database performance. So it will depend mainly on the database performance.

**Comment by giblet22** [comment]
After a conversation with @swise5 and @ramyaragupathy, having a Jupyter Notebook result for different users (ie regular user, project managers, validators, TM overview) would be very useful.

Potential requirements from Quality Control Working Group, an output, (.csv? shared spreadsheet?) of all projects to view:
- project number
- project organization
- associated teams
- percentage done
- percentage validated
- number of total tasks in project
- is project using id presets metadata?
- length of instructions

**Comment by NoemiNahomy** [comment]
I'd like to work on this because I'm working on the other profiling notebook
